hedera-node/hedera-mono-service/src/main/java/com/hedera/node/app/service/mono/stream/RecordStreamManager.java

```diff
@@ -75,22 +79,39 @@ public class RecordStreamManager {
     /** initial running Hash of records */
     private Hash initialHash = new ImmutableHash(new byte[DigestType.SHA_384.digestLength()]);
 
+    /**
+     * If non-null, indicates we are in recovery mode, and the manager should use this writer to ensure
+     * the first generated recovery record includes "prefix" items from any record file on disk whose
+     * 2-second time range includes the first consensus time in the recovery event stream.
+     *
+     * <p>That is, suppose:
+     * <ol>
+     *     <li>There is a record file {@code F} on disk with consensus times in the range {@code [X, X + 2s)}</li>
+     *     <li>The recovery event stream begins at time {@code T}, where {@code T > X} and {@code T < X + 2s}.</li>
+     * </ol>
+     * Then we need to ensure the first record file we generate includes the prefix of items from
+     * {@code F} in the time range {@code [X, T)}.
+     */
```

Comment:
Thank you ! 🙌

Reply:
No, we had just been skipping existing files on disk.

This was a reasonable approach when "replaying" transactions during reconnect; and otherwise we didn't have an easy way to detect whether event stream recovery was happening. 🤔